api/src/main/scala/org/locationtech/sfcurve/SpaceFillingCurve2D.scala

```diff
@@ -11,5 +11,5 @@ package org.locationtech.sfcurve
 trait SpaceFillingCurve2D {
   def toIndex(x: Double, y: Double): Long
   def toPoint(i: Long): (Double, Double)
-  def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double): Seq[(Long, Long)]
+  def toRanges(xmin: Double, ymin: Double, xmax: Double, ymax: Double, maxRecurse: Int): Seq[(Long, Long, Boolean)]
```

Comment:
It might be worth adding a brief comment to explain the 'contains' flag.

A few test cases might make this clear.